### DISCUSSION TITLE
how to only extract the visible text and remove the hidden text from get_text()?

### DISCUSSION THREAD

**Discussion by bk111** [pymupdf/PyMuPDF#2424] [Q&A] [answered]
I extract with get_text(), but I got some hidden text. how to remove the hidden text from get_text()?
the picture and pdf refer to link, 

https://forum.chinese-learning.me/viewtopic.php?f=5&t=412&sid=95ce32a64e7497a19087cdc77bb565c0

**Comment by JorjMcKie** [comment]
Don't understand your post: the page displayed (4) contains no hidden text.

Otherwise, there is method `page.get_bboxlog()` which lists the bboxes within the page, that are filled with some object (text, images, drawings, etc.).
If the page contains text **_declared_** as being hidden (`Tr` operator value 3), then you will find a bbox item looking like `("ignore-text", bbox)`. You can take the bbox of a piece of text as returned by `page.get_text()` and check whether it is contained in some of the ignore-text items.

**Comment by bk111** [comment]
> Don't understand your post: the page displayed (4) contains no hidden text.
> 
> Otherwise, there is method `page.get_bboxlog()` which lists the bboxes within the page, that are filled with some object (text, images, drawings, etc.). If the page contains text **_declared_** as being hidden (`Tr` operator value 3), then you will find a bbox item looking like `("ignore-text", bbox)`. You can take the bbox of a piece of text as returned by `page.get_text()` and check whether it is contained in some of the ignore-text items.

hidden text are inside red circles on the left

**Comment by JorjMcKie** [comment, marked as answer]
> hidden text are inside red circles on the left

Which version are you using? With the current version, that text does not appear.

**Reply by bk111** [reply]
C:\Users\dengz\Downloads\duo>wkhtmltopdf -V
wkhtmltopdf 0.12.6 (with patched qt)

**Reply by bk111** [reply]
PyMuPDF                  1.19.1

**Reply by bk111** [reply]
> > hidden text are inside red circles on the left
> 
> Which version are you using? With the current version, that text does not appear.

after upgrade, it disappear. thanks

**Comment by JorjMcKie** [comment]
Converting to a "Discussions" item.
